openfl/component/director/director.py

```diff
@@ -6,50 +6,282 @@
 import asyncio
 import logging
 import time
-import typing
 from collections import defaultdict
 from pathlib import Path
+from typing import Iterable
+from typing import List
+from typing import Union
 
+from openfl.component import Aggregator
 from openfl.federated import Plan
 from openfl.protocols import director_pb2
+from openfl.transport import AggregatorGRPCServer
 from openfl.utilities.workspace import ExperimentWorkspace
 
 logger = logging.getLogger(__name__)
 
 ENVOY_HEALTH_CHECK_PERIOD = 60  # in seconds
 
 
+class Status:
+    """Experiment's statuses."""
+
+    PENDING = 'PENDING'
+    FINISHED = 'FINISHED'
+    IN_PROGRESS = 'IN_PROGRESS'
+    FAILED = 'FAILED'
+
+
+class Experiment:
+    """Experiment class."""
+
+    def __init__(
+            self, *,
+            name: str,
+            data_path: Union[Path, str],
+            collaborators: List[str],
+            sender: str,
+            init_tensor_dict: dict,
+            plan_path: Union[Path, str] = 'plan/plan.yaml',
+            users: Iterable[str] = None,
+    ) -> None:
+        """Initialize an experiment object."""
+        self.name = name
+        if isinstance(data_path, str):
+            data_path = Path(data_path)
+        self.data_path = data_path
+        self.collaborators = collaborators
+        self.sender = sender
+        self.init_tensor_dict = init_tensor_dict
+        if isinstance(plan_path, str):
+            plan_path = Path(plan_path)
+        self.plan_path = plan_path
+        self.users = set() if users is None else set(users)
+        self.status = Status.PENDING
+        self.__aggregator_grpc_server = None
+
+    @property
+    def aggregator(self) -> Union[Aggregator, None]:
```

Comment:
We should discuss does it make sense to move this functional part somewhere else, for example to the Experiment registry, so this representation could only contain data

Reply:
This will be convenient if we will persist experiments